### PR TITLE
tikzit: init at 2.0

### PR DESCRIPTION
**Pull request by iblech**
###### Motivation for this change

[tikzit](https://tikzit.github.io/) is a graphical editor for TikZ diagrams (for inclusion in LaTeX files). This pull requests adds tikzit to our repository.

The current version of tikzit is 2.0-rc2. I suggest we wait for 2.0 to be released before including tikzit in nixpkgs, to reduce clutter.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by iblech** [comment]
Thank you, Merlin!

TikZiT is indeed GPL3 Plus. The website doesn't communicate this clearly (a fix is [already in the making](https://github.com/tikzit/tikzit.github.io/pull/3)), but the source code comments clearly state so.

**Comment by iblech** [comment]
Dear @mgttlinger, there ust was [a commit](https://github.com/tikzit/tikzit/issues/32) on the TikZiT repository changing the behavior of `make install`. This might affect our Nix build.

I don't have time right now to look into this, but will do so on the weekend. In case you want to go ahead right now, feel free to do so, of course.

**Comment by mgttlinger** [comment]
@iblech Given that we target the released version the new commit won't be in the version we pull, but we should definitely look into that for the next release.

**Comment by iblech** [comment]
@mgttlinger I'm happy to report that we don't need our `preBuild` kludge anymore; just setting the `PREFIX` variable is enough.

**Comment by mgttlinger** [comment]
@iblech That's great, now the build looks really clean. Should we remove the "do not merge" from the title?

**Comment by mgttlinger** [comment]
@iblech That's great, now the build looks really clean. Should we remove the "do not merge" from the title?

**Comment by mgttlinger** [comment]
@iblech That's great, now the build looks really clean. Should we remove the "do not merge" from the title?

**Comment by mgttlinger** [comment]
@iblech That's great, now the build looks really clean. Should we remove the "do not merge" from the title?

**Comment by mgttlinger** [comment]
@iblech That's great, now the build looks really clean. Should we remove the "do not merge" from the title?

**Comment by iblech** [comment]
@mgttlinger From [this commit](https://github.com/tikzit/tikzit/commit/f0c92dbae31b12799d0622b8219d7841ab10464f) I have the feeling that the final version will be released quite soonish. Let's wait a couple more hours and then do it. :-) (Of course updating the referenced git commit hash.)

**Comment by iblech** [comment]
@mgttlinger From [this commit](https://github.com/tikzit/tikzit/commit/f0c92dbae31b12799d0622b8219d7841ab10464f) I have the feeling that the final version will be released quite soonish. Let's wait a couple more hours and then do it. :-) (Of course updating the referenced git commit hash.)

**Comment by iblech** [comment]
@mgttlinger From [this
commit](https://github.com/tikzit/tikzit/commit/f0c92dbae31b12799d0622b8219d7841ab10464f)
I have the feeling that the final version will be released quite soonish. Let's
wait a couple more hours and then do it. :-) (Of course updating the referenced
git commit hash.)

**Comment by iblech** [comment]
@mgttlinger From [this
commit](https://github.com/tikzit/tikzit/commit/f0c92dbae31b12799d0622b8219d7841ab10464f)
I have the feeling that the final version will be released quite soonish. Let's
wait a couple more hours and then do it. :-) (Of course updating the referenced
git commit hash.)

**Comment by iblech** [comment]
I just pushed the necessary changes to that we target the just-released version 2.0.

@mgttlinger, it was a pleasure collaborating with you on this build! :-)

Dear committers, this pull request is no longer work in progress. Please consider it for merging.

File: pkgs/tools/typesetting/tikzit/default.nix
Comment (c0bw3b) [inline pkgs/tools/typesetting/tikzit/default.nix]: `licenses.gpl3Plus` is enough since you have `with stdenv.lib;` at the opening of your meta block

File: pkgs/tools/typesetting/tikzit/default.nix
Comment (c0bw3b) [inline pkgs/tools/typesetting/tikzit/default.nix]: Same reason: `platforms.all`

By the way, have you been able to test the build on Darwin/macOS ?

**Review by c0bw3b — COMMENTED** [review]
Also consider:

* adding `enableParallelBuilding = true;` after checking it doesn't break the build

* there seems to be unit tests available (not sure) :
```nix
  doCheck = true;
  checkTarget = "test";
```

**Comment by mgttlinger** [comment]
Ok, so the make target is the default `check` but it doesn't seem to actually check anything
```log
make: Nothing to be done for 'check'.
```

**Comment by mgttlinger** [comment]
@c0bw3b There seem to be tests if you look at the end of the `tikzit.pro` file but I am unsure how to activate them. I have neither a Mac nor a Win machine at hand for testing. Don't know about @iblech.

**Comment by iblech** [comment]
Thank you for the feedback, @c0bw3b, and thank you for implementing the improvements, @mgttlinger!

I also can't test on Mac or Win, and I also didn't figure out how to activate the tests. The project description file `tikzit.pro` does have a conditional clause for making the resulting Makefile build the test suite, but I failed to find out how to activate this flag. No target of the standard Makefile generated by qmake will compile the test suite, hence in my understanding the `make check` doesn't actually do anything.

**Comment by mgttlinger** [comment]
Note that the travis config in the upstream repo doesn't run those tests either (it seems) so no idea how "maintained" they are anyway.

**Comment by c0bw3b** [comment]
Should be fine, upstream doesn't doesn't document special steps for building on Darwin.
Win is out of scope. Nix mostly targets x64 Linux, aarch64 (ARMv8) and Darwin (macOS)

**Review by c0bw3b — APPROVED** [review]

**Pull request merged**
